### DISCUSSION TITLE
Handling Laravel failed jobs messages

### DISCUSSION THREAD

**Discussion by lucaros97** [inertiajs/inertia#449] [Help] [answered]
Hello,
I'm working in a Laravel app where I call some jobs in a controller.
```php
public function myFunction()
{
    MyJob::dispatch();
    return back();
}
```

If my jobs fail I use the default failed Laravel method which allow me to take some actions. In this case I want to throw an Exception to show in the Vue component the error message:
```php
public function failed(Throwable $exception)
{
    throw MyCustomException::jobFailed();
}
```

And in my Exception I have this:
```php
public static function jobFailed()
{
    return new static("Something went wrong! Check your data and try again");
}
```

Now I jave added a specific response in my `render` function inside the `Handler` class:
```php
public function render($request, Throwable $exception)
{
    if ($exception instanceof CannotAddRegistrar) {
        return back()->withError('Something went wrong');
    }

    return parent::render($request, $exception);
}
```

But nothing seems to be passed in the Inertia props in my Vue component. Any way to get this working?

**Comment by ajnsn** [comment, marked as answer]
@lucaros97 Afaik Laravel's [`RedirectResponse`](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Http/RedirectResponse.php) does not have a `withError()` method.

What you can do is to use the `withErrors()` method and choose a field for your error, e.g. for the login response:

```php
return back()->withErrors(['email' => 'The email and password that you entered did not match our records.']);
```

But this is not too reusable. 
Therefore I often add a property to Inertia's [shared data](https://inertiajs.com/shared-data), e.g. "message".

```php
public function share(Request $request) : array
{
    return array_merge(parent::share($request), [
        'message' => $request->session()->get('message'),
    ]);
}
```

This can be populated by your exception handler like this:

```php
return back()->with('message', 'Something went wrong');
```

In your frontend, you just need the `message` prop and output it somewhere. Or catch it via [`transformProps`](https://inertiajs.com/transforming-props) and show e.g. a instant notification.

**Reply by lucaros97** [reply]
Right, I was already using Inertia's shared data for other stuffs but I was wondering if there's some integrated methods. 
Anyway I found the right solution for this case using an Axios request instead of an inertia request and then use Pusher to throw a message through private channels. 
This became necessary because I had to handle multiple possible errors from 2 jobs called from the same request.

Thank you for you help @ajnsn and hope this can help someone else!
